server/data/tokenStore.ts

```diff
@@ -1,38 +1,45 @@
-import redis from 'redis'
-import { promisify } from 'util'
+import { createClient } from 'redis'
 
 import logger from '../../logger'
 import config from '../config'
 
+const redisURL =
+  config.redis.tls_enabled === 'true'
+    ? `rediss://${config.redis.host}:${config.redis.port}`
+    : `redis://${config.redis.host}:${config.redis.port}`
+
 const createRedisClient = () => {
-  return redis.createClient({
-    port: config.redis.port,
+  return createClient({
+    url: redisURL,
     password: config.redis.password,
-    host: config.redis.host,
-    tls: config.redis.tls_enabled === 'true' ? {} : false,
-    prefix: 'systemToken:',
   })
 }
 
+export type RedisClient = ReturnType<typeof createRedisClient>
+
 export default class TokenStore {
-  private getRedisAsync: (key: string) => Promise<string | null>
+  private readonly prefix = 'systemToken:'
 
-  private setRedisAsync: (key: string, value: string, mode: string, durationSeconds: number) => Promise<void>
+  private readonly redisClient: RedisClient
 
-  constructor(redisClient: redis.RedisClient = createRedisClient()) {
+  constructor(redisClient: RedisClient = createRedisClient()) {
+    this.redisClient = redisClient
     redisClient.on('error', error => {
       logger.error(error, `Redis error`)
     })
-
-    this.getRedisAsync = promisify(redisClient.get).bind(redisClient)
-    this.setRedisAsync = promisify(redisClient.set).bind(redisClient)
   }
 
   public async setToken(key: string, token: string, durationSeconds: number): Promise<void> {
-    await this.setRedisAsync(key, token, 'EX', durationSeconds)
+    if (!this.redisClient.isOpen) {
+      await this.redisClient.connect()
+    }
+    await this.redisClient.set(`${this.prefix}${key}`, token, { EX: durationSeconds })
```

Comment:
I'm tempted to suggest renaming `key` to `username` as this is really what it is, it's the username which is prefixed with `systemToken:` to form the redis _key_ used to set/get the value.

Feel free to leave as it is now tho - I just think `username` is more explicit as these methods are setting/getting the token _for_ the given username.

Reply:
🤔  although now that I think about it, the TokenStore is agnostic of what's stored in it so maybe the fact that `key` is more generic than `username` is actually good.

Thinking "aloud", ignore me I suppose 🤷